autometrics-macros/src/lib.rs

```diff
@@ -377,3 +389,121 @@ histogram_quantile(0.95, {latency})"
 fn concurrent_calls_query(gauge_name: &str, label_key: &str, label_value: &str) -> String {
     format!("sum by (function, module) {gauge_name}{{{label_key}=\"{label_value}\"}}")
 }
+
+fn derive_get_label_impl(input: DeriveInput) -> Result<TokenStream> {
+    let variants = match input.data {
+        Data::Enum(DataEnum { variants, .. }) => variants,
+        _ => {
+            return Err(syn::Error::new_spanned(input, "#[derive(LabelValues}] is only supported for enums"));
+        },
+    };
+
+    let label_key = {
+        let attrs: Vec<_> = input.attrs.iter().filter(|attr| attr.path.is_ident("autometrics")).collect();
+
+        let key_from_attr = match attrs.len() {
+            0 => None,
+            1 => get_label_attr(attrs[0], "label_key")?,
+            _ => {
+                let mut error =
+                    syn::Error::new_spanned(attrs[1], "redundant `autometrics(label_value)` attribute");
+                error.combine(syn::Error::new_spanned(attrs[0], "note: first one here"));
+                return Err(error);
+            }
+        };
+
+        let key_from_attr = key_from_attr.map(|value| value.to_string());
+
+        // Check casing of the user-provided value
+        if let Some(key) = &key_from_attr {
+            if key.as_str() != key.to_snake_case() {
+                return Err(syn::Error::new_spanned(attrs[0], "label_key should be snake_cased"));
+            }
+        }
+
+        let ident = input.ident.clone();
+        key_from_attr.unwrap_or_else(|| ident.clone().to_string().to_snake_case())
+    };
+
+    let value_match_arms = variants
+        .into_iter()
+        .map(|variant| {
+            let attrs: Vec<_> = variant.attrs.iter().filter(|attr| attr.path.is_ident("autometrics")).collect();
+
+            let value_from_attr = match attrs.len() {
+                0 => None,
+                1 => get_label_attr(attrs[0], "label_value")?,
+                _ => {
+                    let mut error =
+                        syn::Error::new_spanned(attrs[1], "redundant `autometrics(label_value)` attribute");
+                    error.combine(syn::Error::new_spanned(attrs[0], "note: first one here"));
+                    return Err(error);
+                }
+            };
+
+            let value_from_attr = value_from_attr.map(|value| value.to_string());
```

Comment:
Do you actually need to turn this into a string in order to use it in quote?

Reply:
I need already need it as a string to be able to detect whether it's snake-cased or not.